api/identifiers/views.py

```diff
@@ -50,6 +48,11 @@ def get_default_queryset(self):
     def get_queryset(self):
         return self.get_queryset_from_request()
 
+    def create(self, *args, **kwargs):
+        obj = self.get_object()
+        if hasattr(obj, 'is_retracted') and obj.is_retracted:
+            raise PermissionDenied(detail='Not allowed to create identifiers for withdrawn preprints or registrations')
```

Comment:
Permission checking belongs in a permission class... how about adding a `ReadOnlyOrNotWithdrawn` to `api.nodes.permissions` that's like [`ExcludeWithdrawals`](https://github.com/CenterForOpenScience/osf.io/blob/5959d3a29f73acc520e2ae7b56d270db8b519072/api/nodes/permissions.py#L116) but returns `request.method in permissions.SAFE_METHODS` instead of `False`?